runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -4758,10 +4765,26 @@ J9::Z::TreeEvaluator::multianewArrayEvaluator(TR::Node * node, TR::CodeGenerator
 
    bool use64BitClasses = comp->target().is64Bit() && !TR::Compiler->om.generateCompressedObjectHeaders();
 
-   // Init class field, then jump to end of ICF
    generateRXInstruction(cg, use64BitClasses ? TR::InstOpCode::STG : TR::InstOpCode::ST, node, classReg, generateS390MemoryReference(targetReg, TR::Compiler->om.offsetOfObjectVftField(), cg));
-   cursor = generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, cFlowRegionEnd);
-   iComment("Init class field and jump to end of ICF.");
+#if defined(TR_TARGET_64BIT)
+   if (isIndexableDataAddrPresent)
+      {
+      TR_ASSERT_FATAL_WITH_NODE(node
+         , IS_32BIT_SIGNED(fej9->getOffsetOfDiscontiguousDataAddrField() - fej9->getOffsetOfContiguousDataAddrField())
+         , "dataAddrFieldOffset is too big for the instruction.");
+
+      // Load dataAddr slot offset difference since 0 size arrays are treated as discontiguous.
+      generateRILInstruction(cg
```

Comment:
If you are choosing to use multi-line statement, please follow how it is done at other places in the code base. You can find multiple example in code base  (e.g, https://github.com/eclipse-openj9/openj9/blob/master/runtime/compiler/optimizer/J9Inliner.cpp)

Reply:
I did it that way to make multi-line statement easier to read. Here's an instance where the benefit is more noticeable  https://github.com/eclipse-openj9/openj9/blob/ea4136ea95e21c50e441b50c3eaa891ca07a1c2f/runtime/compiler/z/codegen/J9TreeEvaluator.cpp#L10893-L10897

Reply:
Could I make a case to have the commas in front?
Having the comma at front would make it easier to differentiate between different assert arguments. It's similar to how having the conditional operators in front rather than at the end aids boolean expression readability.
```c++
TR_ASSERT_FATAL_WITH_NODE(node
      , ((TR::Compiler->om.compressObjectReferences()
            && (fej9->getOffsetOfDiscontiguousDataAddrField() - fej9->getOffsetOfContiguousDataAddrField()) == 8)
        || (!TR::Compiler->om.compressObjectReferences()
            && fej9->getOffsetOfDiscontiguousDataAddrField() == fej9->getOffsetOfContiguousDataAddrField())
      , "Offset of dataAddr field in discontiguous array is expected to be 8 bytes more than contiguous array if using compressed refs, "
        "Or same if using full refs. "
        "But was %d bytes for discontigous and %d bytes for contiguous array.\n"
      , fej9->getOffsetOfDiscontiguousDataAddrField(), fej9->getOffsetOfContiguousDataAddrField());
```
```c++
TR_ASSERT_FATAL_WITH_NODE(node,
      ((TR::Compiler->om.compressObjectReferences()
            && (fej9->getOffsetOfDiscontiguousDataAddrField() - fej9->getOffsetOfContiguousDataAddrField()) == 8)
      || (!TR::Compiler->om.compressObjectReferences()
            && fej9->getOffsetOfDiscontiguousDataAddrField() == fej9->getOffsetOfContiguousDataAddrField()),
      "Offset of dataAddr field in discontiguous array is expected to be 8 bytes more than contiguous array if using compressed refs, "
      "Or same if using full refs. "
      "But was %d bytes for discontigous and %d bytes for contiguous array.\n",
      fej9->getOffsetOfDiscontiguousDataAddrField(), fej9->getOffsetOfContiguousDataAddrField());
```

Reply:
Hmm, I understand the point of making it readable but honestly, I really do not see big difference in the two version you posted (Though my view on Assert is that the condition we are checking before throwing assert should be readable in code which it is in both case). I am just being a nit-picky here, but the reason I made this suggestion was to have it similar to other places in code base.